### PR TITLE
[IOCOM-282] Fixing eslint errors and warnings

### PR DESCRIPTION
**Pull request by adelloste**
## Short description
This PR fixes eslint errors and warnings on behalf of IOCOM team.

### CODE REVIEW

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: src/payloads/services/factory.ts [hunk 1]
```diff
@@ -104,19 +106,16 @@ const createServices = (
...
-): ServicePublic[] => {
-  const services: ServicePublic[] = [];
-  for (
-    // eslint-disable-next-line functional/no-let
-    let serviceIndex = serviceStartIndex;
-    serviceIndex < serviceStartIndex + count;
-    serviceIndex++
-  ) {
+): ServicePublic[] =>
+  A.makeBy(count, count => {
```
[the remark this answers is not in the document]
Comment (adelloste) [reply src/payloads/services/factory.ts]: Applied https://github.com/pagopa/io-dev-api-server/pull/264/commits/583d6a20dd406c0b5c024b571aba01ae3fec243f

File: src/payloads/services/factory.ts [hunk 2]
```diff
@@ -172,29 +168,27 @@ const createSpecialServicesInternal = (
...
-      const specialServiceGenerator = specialServiceGeneratorTuple[1];
-      const specialService = specialServiceGenerator(
+  return pipe(
+    specialServiceGeneratorTuples,
+    A.filter(([isSpecialServiceEnabled, _]) => isSpecialServiceEnabled),
+    A.map(([_, specialServiceGenerator]) => {
+      const organizationFiscalCode = getOrganizationFiscalCode(
+        organizationStartIndex
+      );
+      organizationStartIndex++;
```
[the remark this answers is not in the document]
Comment (adelloste) [reply src/payloads/services/factory.ts]: I got you, you're right. I thought the change was not really relevant however I refactored restoring the old business-logic https://github.com/pagopa/io-dev-api-server/pull/264/commits/9769902d25f6c5a58fbc24e7ce6db6d8889fd38f

File: src/routers/message.ts [hunk 3]
```diff
@@ -45,35 +45,33 @@ const getPublicMessages = (
     }
     const extraData = enrichData
       ? {
+          ...message,
```
[the remark this answers is not in the document]
Comment (adelloste) [reply src/routers/message.ts]: Changed https://github.com/pagopa/io-dev-api-server/pull/264/commits/fe8669c5575bbdde7ba144a4f0889a330634f22e

File: src/populate-persistence.ts [hunk 4]
```diff
@@ -559,132 +545,133 @@ const createMessagesWithPaymentWithValidDueDate = (
...
-          messageMarkdown
+  customConfig: IoDevServerConfig
+): CreatedMessageWithContentAndAttachments[] =>
+  pipe(
+    customConfig.services.specialServices.pn,
+    B.fold(
+      () => [],
+      () =>
+        pipe(
+          customConfig.messages.pnCount,
```
[the remark this answers is not in the document]
Comment (adelloste) [reply src/populate-persistence.ts]: Changed

**Review by Vangaorth — APPROVED** [review]
LGTM

**Pull request merged**